internal/controller/management_backup_controller.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2024
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	velerov1 "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	"k8s.io/client-go/util/workqueue"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	kcmv1alpha1 "github.com/K0rdent/kcm/api/v1alpha1"
+	"github.com/K0rdent/kcm/internal/controller/backup"
+)
+
+// ManagementBackupReconciler reconciles a ManagementBackup object
+type ManagementBackupReconciler struct {
+	client.Client
+
+	config *backup.Config
+
+	SystemNamespace string
+}
+
+func (r *ManagementBackupReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	l := ctrl.LoggerFrom(ctx)
+
+	mgmtBackup := new(kcmv1alpha1.ManagementBackup)
+	if err := r.Client.Get(ctx, req.NamespacedName, mgmtBackup); client.IgnoreNotFound(err) != nil {
+		l.Error(err, "unable to fetch ManagementBackup")
+		return ctrl.Result{}, err
+	}
+
+	mgmt, err := r.config.GetManagement(ctx)
+	if err != nil && !errors.Is(err, backup.ErrNoManagementExists) { // error during list
+		return ctrl.Result{}, err
+	}
+
+	if errors.Is(err, backup.ErrNoManagementExists) || !mgmt.Spec.Backup.Enabled {
+		if mgmtBackup.IsSchedule() {
+			if !mgmtBackup.Status.Paused {
+				mgmtBackup.Status.Paused = true
+				if err := r.Client.Status().Update(ctx, mgmtBackup); err != nil {
+					return ctrl.Result{}, fmt.Errorf("failed to mark ManagementBackup status as paused: %w", err)
+				}
+			}
+		}
+
+		l.V(1).Info("No Management object exists or backup is disabled, nothing to do")
+		return ctrl.Result{}, nil
+	}
+
+	if mgmtBackup.CreationTimestamp.IsZero() || mgmtBackup.UID == "" { // mgmt exists
+		requestEqualsMgmt := mgmt != nil && req.Name == mgmt.Name && req.Namespace == mgmt.Namespace
```

Comment:
`req.Namespace == mgmt.Namespace` 
Management and ManagementBackup objects are cluster-scoped. Do we need this condition?

Reply:
for consistency, but yeah it is not required so I dropped it, resolving this and the same comment below